### DISCUSSION TITLE
Manage Pipelines-> Pipelines not detected

### DISCUSSION THREAD

**Discussion by AI-ANK** [open-webui/open-webui#4232] [General]
# Bug Report

## Description

**Bug Summary:**
I am unable to see the option to upload pipelines, in admin panel under "Manage Pipelines"
Earlier it used to work, but after a fresh install it is no longer wokring. I have check under connections and the openai and piplines connection are working fine. Any idea what am I missing?
Current screenshot: ![image](https://github.com/user-attachments/assets/23b20675-ea39-4557-b485-84f562d6b4cd)

**Comment by hpsc3** [comment]
Any response on this?

**Reply by justinh-rahb** [reply]
Is the API key for your pipelines server set to the default of `0p3n-w3bu!` ?

**Comment by TimoKrocker** [comment]
When you are using docker, perhaps you forgot to start the docker container for the  pipelines.

**Comment by chuckjewell** [comment]
This issue is still going on.

I have a new installation of openwebui and pipeline docker containers on a fresh OS installed mac M3. 

I used the openwebui docker command with ollama integration from the project README.md

Openwebui runs great, sees my ollama models. I can add the pipeline connection and verify it. Note I have tried localhost, docker internal and currently the actual IP address from the container. All three connection tests pass

Pipelines still says: Pipelines Not Detected

Looking at the docker console, I see there are a few failures in the container. Are these expected?

2025-01-22 13:59:59 RESET_PIPELINES_DIR is not set to true. No action taken.
2025-01-22 13:59:59 PIPELINES_REQUIREMENTS_PATH not specified. Skipping installation of requirements.
2025-01-22 13:59:59 PIPELINES_URLS not specified. Skipping pipelines download and installation.
2025-01-22 14:00:04 INFO:     172.17.0.2:33278 - "GET /models HTTP/1.1" 200 OK
2025-01-22 14:00:08 INFO:     172.17.0.2:33294 - "GET /models HTTP/1.1" 200 OK
2025-01-22 14:00:09 INFO:     172.17.0.2:57250 - "GET /pipelines HTTP/1.1" 200 OK
2025-01-22 14:00:11 INFO:     172.17.0.2:57266 - "GET /models HTTP/1.1" 200 OK
2025-01-22 14:00:21 INFO:     172.17.0.2:56246 - "GET /models HTTP/1.1" 200 OK
2025-01-22 14:00:00 INFO:     Started server process [7]
2025-01-22 14:00:00 INFO:     Waiting for application startup.
2025-01-22 14:00:00 INFO:     Application startup complete.
2025-01-22 14:00:00 INFO:     Uvicorn running on http://0.0.0.0:9099 (Press CTRL+C to quit)

**Reply by bielids** [reply]
I have the same problem, but in my case I don't even see any calls being made to /pipelines. One thing I noticed is that even though I had disabled other OpenAI connections, they were still being queried by Open WebUI's frontend when I would click on the Pipelines link in the admin UI. After deleting these connections entirely I now see new debug messages which indicate that Open WebUI's frontend isn't able to find models.

pipelines container logs: 
```
RESET_PIPELINES_DIR is not set to true. No action taken.
PIPELINES_REQUIREMENTS_PATH not specified. Skipping installation of requirements.
PIPELINES_URLS not specified. Skipping pipelines download and installation.
INFO:     Started server process [7]
INFO:     Waiting for application startup.
INFO:     Application startup complete.
INFO:     Uvicorn running on http://0.0.0.0:9099 (Press CTRL+C to quit)
Loaded module: mycustompipeline
on_startup:blueprints.function_calling_blueprint
INFO:     172.28.5.242:54374 - "GET /models HTTP/1.1" 200 OK
INFO:     172.28.5.242:47374 - "GET /models HTTP/1.1" 200 OK
INFO:     172.28.5.242:46050 - "GET /models HTTP/1.1" 200 OK
INFO:     172.28.5.242:44248 - "GET /models HTTP/1.1" 200 OK
INFO:     172.28.5.242:53682 - "GET /models HTTP/1.1" 200 OK
```

and when I check Open WebUI's container logs I see this upon click on the Pieplines page: 
```
ERROR [open_webui.routers.openai] Connection error: 403, message='Attempt to decode JSON with unexpected mimetype: text/html;charset=utf-8', url='http://openwebui-pipelines:9099/models'
DEBUG [open_webui.routers.openai] get_all_models:responses() [None]
DEBUG [open_webui.routers.pipelines] get_pipelines_list: get_openai_models_responses returned [None]
```

If I query the pipelines API manually via curl I do see a model though.
```
└─➜ curl  -H "Authorization: Bearer 0p3n-w3bu\!" openwebui-pipelines:9099/models                                                                                                 [0]
{"data":[{"id":"mycustompipeline","name":"My Tools Pipeline","object":"model","created":1737645129,"owned_by":"openai","pipeline":{"type":"filter","pipelines":["*"],"priority":0,"valves":true}}],"object":"list","pipelines":true}%  
```

and the connection check in the admin panel passes (so the API key is correct) 

I'm pulled the `main` tag for both the openwebui and pipelines containers this morning.

**Reply by bielids** [reply]
I figured out my problem. My request was mistakenly being routed via a Squid proxy, hence the wrong mimetype.
